python-sdk/nuscenes/predict/tests/run_covernet.py

```diff
@@ -0,0 +1,94 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+"""
+Regression test to see if CoverNet implementation can overfit on a single example.
+"""
+
+import argparse
+
+import math
+import numpy as np
+import torch
+import torch.optim as optim
+from torch.utils.data import DataLoader, IterableDataset
+
+from nuscenes.predict.models.backbone import MobileNetBackbone
+from nuscenes.predict.models.covernet import CoverNet, ConstantLatticeLoss
+
+
+def generate_trajectory(theta: float) -> torch.Tensor:
+    trajectory = torch.zeros(6, 2)
+    trajectory[:, 0] = torch.arange(6) * math.cos(theta)
+    trajectory[:, 1] = torch.arange(6) * math.sin(theta)
+    return trajectory
+
+
+class Dataset(IterableDataset):
+    """
+    Implements an infinite dataset
```

Comment:
What is the content of the infinite dataset? Is it always the same?

Reply:
Yes. I've elaborated a bit in the doc string.